infrastructure/templates/template.json

```diff
@@ -997,16 +997,54 @@
         "criterionType": "StaticThresholdCriterion"
       },
       {
-        "name": "[concat(parameters('subscription'),'DataFactoryPipelineFailures')]",
-        "description": "Data Factory - Pipeline Failed Run",
-        "resourceId": "[resourceId('Microsoft.DataFactory/factories', variables('dataFactoryName'))]",
-        "resourceType": "Microsoft.DataFactory/factories",
-        "metricName": "PipelineFailedRuns",
-        "threshold": 1,
-        "operator": "GreaterThan",
-        "timeAggregation": "Total",
-        "criterionType": "StaticThresholdCriterion"
-      }
+        "type": "microsoft.insights/metricAlerts",
```

Comment:
This isn't going to work inside the `metricAlerts` array. This is because this array is looped through elsewhere in the template - you can see it around line 4000:

```
{
      "type": "microsoft.insights/metricAlerts",
      "apiVersion": "2018-03-01",
      "name": "[variables('metricAlerts')[copyIndex()].name]",
      "location": "global",
      "dependsOn": [
        "[variables('metricAlerts')[copyIndex()].resourceId]",
        "[resourceId('microsoft.insights/actionGroups', variables('actionGroupAlerts'))]"
      ],
      "copy": {
        "name": "metricAlertsCopy",
        "count": "[length(variables('metricAlerts'))]"
      },
      "properties": {
        "description": "[variables('metricAlerts')[copyIndex()].description]",
        "severity": 3,
        "enabled": "[parameters('enableAlerts')]",
        "scopes": [
          "[variables('metricAlerts')[copyIndex()].resourceId]"
        ],
        "evaluationFrequency": "PT1M",
        "windowSize": "PT5M",
        "criteria": {
          "allOf": [
            {
              "threshold": "[variables('metricAlerts')[copyIndex()].threshold]",
              "name": "Metric1",
              "metricNamespace": "[variables('metricAlerts')[copyIndex()].resourceType]",
              "metricName": "[variables('metricAlerts')[copyIndex()].metricName]",
              "operator": "[variables('metricAlerts')[copyIndex()].operator]",
              "timeAggregation": "[variables('metricAlerts')[copyIndex()].timeAggregation]",
              "criterionType": "[variables('metricAlerts')[copyIndex()].criterionType]"
            }
          ],
          "odata.type": "Microsoft.Azure.Monitor.MultipleResourceMultipleMetricCriteria"
        },
        "autoMitigate": true,
        "targetResourceType": "[variables('metricAlerts')[copyIndex()].resourceType]",
        "actions": [
          {
            "actionGroupId": "[resourceId('microsoft.insights/actionGroups', variables('actionGroupAlerts'))]"
          }
        ]
      }
    }
```

That uses the `metricAlerts` variable to create all the metric alerts without having to type all that stuff out for each individual metric - I think `copy` tells it what numbers to loop up to, and `copyIndex()` gets the index of the current element in the metricAlerts variable array.

You could move what you've added here below the "microsoft.insights/metricAlerts" instance around line 4000.